src/plugins/chart_expressions/expression_xy/common/types/expression_functions.ts

```diff
@@ -0,0 +1,241 @@
+   */
+  position: Position;
+  /**
+   * Flag whether the legend should be shown even with just a single series
+   */
+  showSingleSeries?: boolean;
+  /**
+   * Flag whether the legend is inside the chart
+   */
+  isInside?: boolean;
+  /**
+   * Horizontal Alignment of the legend when it is set inside chart
+   */
+  horizontalAlignment?: HorizontalAlignment;
+  /**
+   * Vertical Alignment of the legend when it is set inside chart
+   */
+  verticalAlignment?: VerticalAlignment;
+  /**
+   * Number of columns when legend is set inside chart
+   */
+  floatingColumns?: number;
+  /**
+   * Maximum number of lines per legend item
+   */
+  maxLines?: number;
+
+  /**
+   * Flag whether the legend items are truncated or not
+   */
+  shouldTruncate?: boolean;
+
+  /**
+   * Exact legend width (vertical) or height (horizontal)
+   * Limited to max of 70% of the chart container dimension Vertical legends limited to min of 30% of computed width
+   */
+  legendSize?: number;
+}
+
+export interface LabelsOrientationConfig {
+  x: number;
+  yLeft: number;
+  yRight: number;
+}
+
+// Arguments to XY chart expression, with computed properties
+export interface XYArgs {
+  title?: string;
+  description?: string;
+  xTitle: string;
+  yTitle: string;
+  yRightTitle: string;
+  yLeftExtent: AxisExtentConfigResult;
+  yRightExtent: AxisExtentConfigResult;
+  legend: LegendConfigResult;
+  valueLabels: ValueLabelMode;
+  layers: XYLayerConfigResult[];
+  endValue?: EndValue;
+  emphasizeFitting?: boolean;
+  fittingFunction?: FittingFunction;
+  axisTitlesVisibilitySettings?: AxisTitlesVisibilityConfigResult;
+  tickLabelsVisibilitySettings?: TickLabelsConfigResult;
+  gridlinesVisibilitySettings?: GridlinesConfigResult;
+  labelsOrientation?: LabelsOrientationConfigResult;
+  curveType?: XYCurveType;
+  fillOpacity?: number;
+  hideEndzones?: boolean;
+  valuesInLegend?: boolean;
+  ariaLabel?: string;
+}
+
+export interface AnnotationLayerArgs {
+  annotations: EventAnnotationOutput[];
+  layerId: string;
+  hide?: boolean;
+}
+
+export type AnnotationLayerConfigResult = AnnotationLayerArgs & {
+  type: typeof ANNOTATION_LAYER;
+  layerType: typeof LayerTypes.ANNOTATIONS;
+};
+
+export interface ReferenceLineLayerArgs {
+  layerId: string;
+  accessors: string[];
+  columnToLabel?: string;
+  yConfig?: YConfigResult[];
+}
+
+export type XYLayerArgs = DataLayerArgs | ReferenceLineLayerArgs | AnnotationLayerArgs;
+
+export type XYLayerConfigResult =
+  | DataLayerConfigResult
+  | ReferenceLineLayerConfigResult
+  | AnnotationLayerConfigResult;
+
+export interface LensMultiTable {
+  type: typeof MULTITABLE;
+  tables: Record<string, Datatable>;
+  dateRange?: {
+    fromDate: Date;
+    toDate: Date;
+  };
+}
+
+export type ReferenceLineLayerConfigResult = Omit<ReferenceLineLayerArgs, 'yConfig'> & {
+  type: typeof REFERENCE_LINE_LAYER;
+  layerType: typeof LayerTypes.REFERENCELINE;
+  yConfig?: YConfigResult[];
```

Comment:
why omitting `yConfig` from Args and then adding it again? Aren't they the same? 🤔

Reply:
Before it had meaning, but now it is not. Thanks for this remark)

Reply:
Done.